src/compiler/checker.ts

```diff
@@ -12191,9 +12192,7 @@ namespace ts {
                             }
                         }
                     }
-                    if (elementTypes.length) {
-                        return createTupleType(elementTypes);
-                    }
+                    return createTupleType(elementTypes);

```

Comment:
Agreed. We need to make the condition persistent over future updates. A fix is pushed!